src/main/java/com/javaschool/domainlogic/admin/productmanagment/service/impl/GenreServiceImpl.java

```diff
@@ -0,0 +1,77 @@
+package com.javaschool.domainlogic.admin.productmanagment.service.impl;
+
+import com.javaschool.dao.api.product.GenreRepository;
+import com.javaschool.domainlogic.admin.productmanagment.dto.GenreDto;
+import com.javaschool.domainlogic.admin.productmanagment.exception.GenreNotFound;
+import com.javaschool.domainlogic.admin.productmanagment.mapper.GenreMapper;
+import com.javaschool.entity.product.Genre;
+import com.javaschool.domainlogic.admin.productmanagment.service.api.GenreService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class GenreServiceImpl implements GenreService {
+
+    @Autowired
+    private GenreMapper genreMapper;
+
+    @Autowired
+    private GenreRepository genreRepository;
+
+    @Override
+    @Transactional
+    public List<GenreDto> getGenreDtoList() {
+        List<Genre> genreList = genreRepository.findAll();
+        return genreMapper.toDtoList(genreList);
+    }
+
+    @Override
+    @Transactional
```

Comment:
Use @Transactional(readOnly = true) for methods that only get information without changing anything